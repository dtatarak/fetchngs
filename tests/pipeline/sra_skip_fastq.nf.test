nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test,docker"
    tag "pipeline"

    test("SRA Skip FASTQ") {

        when {
            params {
                outdir              = "$outputDir"
                skip_fastq_download = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/metadata/").list()).match("metadata") },
                { assert snapshot(path("$outputDir/samplesheet/").list()).match("samplesheet") },
                { assert !new File("$outputDir/fastq/DRX024467_DRR026872.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/DRX026011_DRR028935_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/DRX026011_DRR028935_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1188904_ERR1109373_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1188904_ERR1109373_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1234253_ERR1160846.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX10940790_SRR14593545_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX10940790_SRR14593545_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX11047067_SRR14709033.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX6725035_SRR9984183.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9315476_SRR12848126_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9315476_SRR12848126_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9626017_SRR13191702_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9626017_SRR13191702_2.fastq.gz").exists() }
            )
        }

    }

}
