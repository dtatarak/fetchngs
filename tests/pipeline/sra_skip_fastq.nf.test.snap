{
    "metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,f46819ed848d0d28dff80fd0f5a4cd34",
            "DRR028935.runinfo_ftp.tsv:md5,ef08c3756034e6705b0e13d1f37bfea5",
            "ERR1109373.runinfo_ftp.tsv:md5,6e7960f43046c6080435a024e9c426c9",
            "ERR1160846.runinfo_ftp.tsv:md5,db8b7bc5877a62b90e719d7cf3c0d25f",
            "SRR12848126.runinfo_ftp.tsv:md5,b8ac151eb85a001e9aa3d7319156c016",
            "SRR13191702.runinfo_ftp.tsv:md5,7188ac3b1d73ec297a900450e73bf751",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,827751e211294eeecbb7239f70ead9a5"
        ],
        "timestamp": "2023-05-09T00:28:56+0000"
    },
    "samplesheet": {
        "content": [
            "id_mappings.csv:md5,40c6daa8d668b4930bfd9045248e16ec",
            "multiqc_config.yml:md5,a6370add94bbe60c6bbe1bb48a038e8d",
            "samplesheet.csv:md5,2006cc16ca44e947b5244528310c03e0"
        ],
        "timestamp": "2023-05-09T00:28:56+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,8b5ec8d90481271f6584ceff4d990caa"
        ],
        "timestamp": "2023-05-09T00:28:56+0000"
    }
}